pkg/controller/jobs/leaderworkerset/leaderworkerset_webhook.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package leaderworkerset
+
+import (
+	"context"
+	"fmt"
+
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	leaderworkersetv1 "sigs.k8s.io/lws/api/leaderworkerset/v1"
+
+	"sigs.k8s.io/kueue/pkg/controller/constants"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/controller/jobs/pod"
+)
+
+type Webhook struct {
+	client                     client.Client
+	manageJobsWithoutQueueName bool
+}
+
+func SetupWebhook(mgr ctrl.Manager, opts ...jobframework.Option) error {
+	options := jobframework.ProcessOptions(opts...)
+	wh := &Webhook{
+		client:                     mgr.GetClient(),
+		manageJobsWithoutQueueName: options.ManageJobsWithoutQueueName,
+	}
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&leaderworkersetv1.LeaderWorkerSet{}).
+		WithDefaulter(wh).
+		WithValidator(wh).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-apps-v1-leaderworkerset,mutating=true,failurePolicy=fail,sideEffects=None,groups="apps",resources=leaderworkersets,verbs=create,versions=v1,name=mleaderworkerset.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomDefaulter = &Webhook{}
+
+func (wh *Webhook) Default(ctx context.Context, obj runtime.Object) error {
+	lws := fromObject(obj)
+	log := ctrl.LoggerFrom(ctx).WithName("leaderworkerset-webhook")
+	log.V(5).Info("Applying defaults")
+
+	cqLabel, ok := lws.Labels[constants.QueueLabel]
+	if !ok {
+		return nil
+	}
+
+	// LeaderTemplate defaults to the WorkerTemplate if not specified
+	if lws.Spec.LeaderWorkerTemplate.LeaderTemplate != nil {
```

Comment:
What if user updates this field and sets it to nil? Maybe we should prevent this case in the Update webhook.